vernac/vernacentries.ml

```diff
@@ -1185,9 +1185,15 @@ let vernac_hints ~atts dbnames h =
       (warn_implicit_core_hint_db (); ["core"])
     else dbnames
   in
-  let local, poly = Attributes.(parse Notations.(locality ++ polymorphic) atts) in
-  let local = enforce_module_locality local in
-  Hints.add_hints ~local dbnames (Hints.interp_hints ~poly h)
+  let locality, poly = Attributes.(parse Notations.(option_locality ++ polymorphic) atts) in
+  let () = match locality with
+  | OptGlobal ->
+    if Global.sections_are_opened () then
+    CErrors.user_err Pp.(str
+      "This command does not support the Global option in sections.");
```

Comment:
```suggestion
      "This command does not support the global attribute in sections.");
```

Also, not sure this should be an error given that the usual unsupported attribute message is just a fatal-by-default warning (OK this was already an error message so you don't want to change this).

Also, why is this check only performed for `global` and not for `export`?

Finally, this was the only place where the `enforce_module_locality` function was used, so please remove it from `vernac/locality.ml{,i}`.

Reply:
Tweaked the message, and added the check for export as well. I didn't remove `enforce_module_locality` though, as it might be used by plugins, and furthermore it would make the Locality API kind of asymmetrical, since all other localities come as pairs `make` / `enforce`. I'd rather remove this API at once instead.

Reply:
OK that makes sense.